src/Map/Map.tsx

```diff
@@ -0,0 +1,166 @@
+  const pointColor = props.clusterOptions?.pointColor || colors.black
+  const pointRadius = props.clusterOptions?.pointRadius || 6
+
+  useEffect(() => {
+    if (map.current) return // initialize map only once
+    map.current = new mapboxgl.Map({
+      container: mapContainer.current!,
+      style: style,
+      center: [long, lat],
+      zoom: zoom,
+      attributionControl: false,
+    })
+  })
+
+  useEffect(() => {
+    if (!map.current) return // wait for map to initialize
+    map.current.on('load', () => {
+      map.current?.addSource('source', {
+        type: 'geojson',
+        data: source,
+        cluster: cluster,
+        clusterMaxZoom: clusterMaxZoom, // Max zoom to cluster points on
+        clusterRadius: clusterAreaRadius, // Radius of each cluster when clustering points (defaults to 50)
+      })
+
+      map.current?.addLayer({
+        id: 'clusters',
+        type: 'circle',
+        source: 'source',
+        filter: ['has', 'point_count'],
+        paint: {
+          'circle-color': clusterColor,
+          'circle-radius': clusterRadius,
+        },
+      })
+
+      map.current?.addLayer({
+        id: 'cluster-count',
+        type: 'symbol',
+        source: 'source',
+        filter: ['has', 'point_count'],
+        layout: {
+          'text-field': ['get', 'point_count_abbreviated'],
+          'text-font': countFont,
+          'text-size': countFontSize,
+        },
+        paint: {
+          'text-color': countFontColor,
+        },
+      })
+
+      map.current?.addLayer({
+        id: 'unclustered-point',
+        type: 'circle',
+        source: 'source',
+        filter: ['!', ['has', 'point_count']],
+        paint: {
+          'circle-color': pointColor,
+          'circle-radius': pointRadius,
+        },
+      })
+
+      // inspect a cluster on click
+      map.current?.on('click', 'clusters', (e) => {
+        const features = map.current!.queryRenderedFeatures(e.point, {
+          layers: ['clusters'],
+        })
+
+        const clusterId = features[0]?.properties?.cluster_id
+        const source: mapboxgl.GeoJSONSource = map.current!.getSource(
+          'source'
+        ) as mapboxgl.GeoJSONSource
+        source.getClusterExpansionZoom(clusterId, (err, zoom) => {
+          if (err) return
+
+          if (features[0].geometry.type === 'Point') {
+            const coordinates = features[0].geometry.coordinates
+            map.current?.easeTo({
+              center: coordinates as mapboxgl.LngLatLike,
+              zoom: zoom,
+            })
+          }
+        })
+      })
+
+      map.current?.on('mouseenter', 'clusters', () => {
+        map.current!.getCanvas().style.cursor = 'pointer'
+      })
+      map.current?.on('mouseleave', 'clusters', () => {
+        map.current!.getCanvas().style.cursor = ''
+      })
+    })
+  })
+
+  return <Wrapper ref={mapContainer} height={height} width={width}></Wrapper>
```

Comment:
maybe a jest test against snapshot?

Reply:
I've managed to load in envs fine with `"test": "jest --setupFiles dotenv/config"`, but the Map webgl doesn't load in jest